package/MDAnalysis/coordinates/chain.py

```diff
@@ -353,6 +353,16 @@ def __init__(self, filenames, skip=1, dt=None, continuous=False, **kwargs):
         self.ts = None
         self.rewind()
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.pop('_ChainReader__chained_trajectories_iter', None)
```

Comment:
So I can't really reproduce the state of this generator cleanly (or at least in a simple way).  Tbh I've never used chainreader or really touched the code, so maybe I should just slap a 'NotImplementedError' on this and let someone who understands this fix it in the future.

Reply:
I think what you're doing is ok — like all the others, it resets to frame 0 on unpickling. 

Once pickle support is upgraded to also restore the frame index then moving to the appropriate frame should just use the usual `trajectory[frame]` and then let ChainReader take care of how to do it.

If you somehow managed to pickle `ChainReader.readers` in https://github.com/MDAnalysis/mdanalysis/blob/6d6b86202a16bff69fad3d8017aca19eed76578c/package/MDAnalysis/coordinates/chain.py#L265-L266 then it should work. Otherwise, we need a special getstate/setstate pair that blanks `readers` and rebuilds them from `ChainReader.filenames` (although we should actually store the original `filenames` because they can be a list of tuples — that's an addition to ChainReader itself).